pkg/autogen/autogen.go

```diff
@@ -240,3 +242,96 @@ func GenerateRulePatches(spec *kyverno.Spec, controllers string, log logr.Logger
 	}
 	return
 }
+
+type Policy interface {
+	GetAnnotations() map[string]string
+	GetSpec() kyverno.Spec
+}
+
+// GenerateRules generates rule for podControllers based on scenario A and C
```

Comment:
What are scenarios A and C?

Reply:
See https://github.com/kyverno/kyverno/blob/3cf83bc77f5eaee09e186372c169efe4dc3927f5/pkg/policymutation/policymutation.go#L241-L248
I'll rework the comment here

Reply:
Done.